sw/device/lib/dif/dif_sysrst_ctrl_unittest.cc

```diff
@@ -240,5 +240,391 @@ TEST_F(UlpWakeupConfigTest, Success) {
+  EXPECT_READ32(SYSRST_CTRL_REGWEN_REG_OFFSET, 0);
+  EXPECT_EQ(dif_sysrst_ctrl_pins_set_inverted(&sysrst_ctrl_, _pins, true),
+            kDifLocked);
+}
+
+TEST_F(PinsSetInvertedTest, Success) {
+  EXPECT_READ32(SYSRST_CTRL_REGWEN_REG_OFFSET, 1);
+  EXPECT_READ32(SYSRST_CTRL_KEY_INVERT_CTL_REG_OFFSET, 0);
+  EXPECT_WRITE32(SYSRST_CTRL_KEY_INVERT_CTL_REG_OFFSET, _pins);
+  EXPECT_DIF_OK(dif_sysrst_ctrl_pins_set_inverted(&sysrst_ctrl_, _pins, true));
+
+  EXPECT_READ32(SYSRST_CTRL_REGWEN_REG_OFFSET, 1);
+  EXPECT_READ32(SYSRST_CTRL_KEY_INVERT_CTL_REG_OFFSET,
+                _pins | kDifSysrstCtrlPinBatteryDisableOut);
+  EXPECT_WRITE32(SYSRST_CTRL_KEY_INVERT_CTL_REG_OFFSET,
+                 kDifSysrstCtrlPinBatteryDisableOut);
+  EXPECT_DIF_OK(dif_sysrst_ctrl_pins_set_inverted(&sysrst_ctrl_, _pins, false));
+}
+
+class PinsGetInvertedTest : public SysrstCtrlTest {};
+
+TEST_F(PinsGetInvertedTest, NullArgs) {
+  uint32_t inverted_pins;
+  EXPECT_DIF_BADARG(dif_sysrst_ctrl_pins_get_inverted(nullptr, &inverted_pins));
+  EXPECT_DIF_BADARG(dif_sysrst_ctrl_pins_get_inverted(&sysrst_ctrl_, nullptr));
+  EXPECT_DIF_BADARG(dif_sysrst_ctrl_pins_get_inverted(nullptr, nullptr));
+}
+
+TEST_F(PinsGetInvertedTest, Success) {
+  uint32_t inverted_pins;
+  EXPECT_READ32(SYSRST_CTRL_KEY_INVERT_CTL_REG_OFFSET,
+                kDifSysrstCtrlPinKey0In | kDifSysrstCtrlPinKey2Out);
+  EXPECT_DIF_OK(
+      dif_sysrst_ctrl_pins_get_inverted(&sysrst_ctrl_, &inverted_pins));
+  EXPECT_EQ(inverted_pins, kDifSysrstCtrlPinKey0In | kDifSysrstCtrlPinKey2Out);
+}
+
+class PinOverrideSetAllowedTest : public SysrstCtrlTest {};
+
+TEST_F(PinOverrideSetAllowedTest, NullArgs) {
+  EXPECT_DIF_BADARG(dif_sysrst_ctrl_output_pin_override_set_allowed(
+      nullptr, kDifSysrstCtrlPinKey1Out, true, true));
+}
+
+TEST_F(PinOverrideSetAllowedTest, BadPin) {
+  EXPECT_DIF_BADARG(dif_sysrst_ctrl_output_pin_override_set_allowed(
+      &sysrst_ctrl_, kDifSysrstCtrlPinKey1In, true, true));
+}
+
+TEST_F(PinOverrideSetAllowedTest, Locked) {
+  EXPECT_READ32(SYSRST_CTRL_REGWEN_REG_OFFSET, 0);
+  EXPECT_EQ(dif_sysrst_ctrl_output_pin_override_set_allowed(
+                &sysrst_ctrl_, kDifSysrstCtrlPinKey1Out, true, true),
+            kDifLocked);
+}
+
+TEST_F(PinOverrideSetAllowedTest, Success) {
+  EXPECT_READ32(SYSRST_CTRL_REGWEN_REG_OFFSET, 1);
+  EXPECT_READ32(SYSRST_CTRL_PIN_ALLOWED_CTL_REG_OFFSET,
+                {{SYSRST_CTRL_PIN_ALLOWED_CTL_PWRB_OUT_0_BIT, 1}});
+  EXPECT_WRITE32(SYSRST_CTRL_PIN_ALLOWED_CTL_REG_OFFSET,
+                 {{SYSRST_CTRL_PIN_ALLOWED_CTL_PWRB_OUT_0_BIT, 1},
+                  {SYSRST_CTRL_PIN_ALLOWED_CTL_EC_RST_L_0_BIT, 1},
+                  {SYSRST_CTRL_PIN_ALLOWED_CTL_EC_RST_L_1_BIT, 1}});
+  EXPECT_DIF_OK(dif_sysrst_ctrl_output_pin_override_set_allowed(
+      &sysrst_ctrl_, kDifSysrstCtrlPinEcResetInOut, true, true));
+}
+
+class PinOverrideGetAllowedTest : public SysrstCtrlTest {
+ protected:
+  bool _allow_zero;
```

Comment:
I don't think the test fixture is the right place for these variables, personally. It's not consistent with how we do most other things at lesat.

Reply:
Ah yes, good catch.